src/ui/RealmBrowser/Content/CreateObjectDialog/types/ObjectIdControl.tsx

```
@@ -0,0 +1,105 @@
+import { parseObjectId } from '../../../parsers';
+
+interface IObjectIdControlState {
+  internalValue: string | null;
+}
+
+export class ObjectIdControl extends React.PureComponent<
+  IBaseControlProps,
+  IObjectIdControlState
+> {
+  constructor(props: IBaseControlProps) {
+    super(props);
+
+    this.state = {
+      internalValue: props.value,
+    };
+  }
+
+  render() {
+    const { children, property, value } = this.props;
+    const { internalValue } = this.state;
+
+    return (
+      <InputGroup className="CreateObjectDialog__ObjectIdControl">
+        <Input
+          className="CreateObjectDialog__ObjectIdControl__Input"
+          onChange={this.inputChangeEventHandler}
+          placeholder={value === null ? 'null' : ''}
+          value={internalValue ?? ''}
+          required={!property.optional}
+          invalid={(!!internalValue || !property.optional) && value === null}
+        />
+        {internalValue && property.optional ? (
+          <InputGroupAddon addonType="append">
+            <Button size="sm" onClick={this.clearValue}>
+              <i className="fa fa-close" />
+            </Button>
+          </InputGroupAddon>
+        ) : (
+          <InputGroupAddon addonType="append">
+            <Button size="sm" onClick={this.generateObjectId}>
+              <i className="fa fa-refresh" />
+            </Button>
+          </InputGroupAddon>
+        )}
+        {children}
+      </InputGroup>
+    );
+  }
+
+  private inputChangeEventHandler = (e: React.ChangeEvent<HTMLInputElement>) =>
+    this.internalChangeHandler(e.target.value);
+
+  private clearValue = () => this.internalChangeHandler(null);
+
+  private internalChangeHandler = (val: string | null) => {
+    const { property, onChange } = this.props;
+
+    this.setState({ internalValue: val });
+
+    let parsedId: ObjectId | null = null;
+
+    if (val) {
+      try {
+        parsedId = parseObjectId(val, property);
+      } catch (_) {
+        // ignored
```

I suggest logging this as a `console.warn`, since the error could be something else than a parsing error. Ex an error in the implementation of `parseObjectId`, which would be hard to find when the error is swallowed completely.
I believe we've done that elsewhere.